### PR TITLE
Consolidate import export comma styles

### PR DESCRIPTION
**Pull request by 3kyro**
Consolidate `import-export-comma-style` and `diff-friendly-import-export` into a new option `import-export-style` as per #201

cc @brandonchinn178

### CODE REVIEW

File: src/Ormolu/Printer/Meat/Module.hs [hunk 1]
```diff
@@ -23,6 +23,8 @@ import Ormolu.Printer.Meat.Declaration.Warning
 import Ormolu.Printer.Meat.ImportExport
 import Ormolu.Printer.Meat.Pragma
 
+{- HLINT ignore "Use camelCase" -}
```
Comment (brandonchinn178) [inline src/Ormolu/Printer/Meat/Module.hs:26]: Can we avoid adding hlint stuff to the repo? We are explicitly not hlint-compatible (see DEVELOPER.md)

File: src/Ormolu/Config/Types.hs [hunk 2]
```diff
@@ -43,3 +45,22 @@ data HaddockPrintStyle
...
+
+data ImportExportStyle
+  = ImportExportLeading
+  | ImportExportTrailing
+  | ImportExportDiffFriendly
+  deriving (Eq, Show, Enum, Bounded)
+
+-- poImportExportStyle helpers.
+
+data DiffFriendly = DiffFriendly
```
Comment (brandonchinn178) [inline src/Ormolu/Config/Types.hs:57]: this doesnt seem to be used?

File: src/Ormolu/Config/Types.hs [hunk 3]
```diff
@@ -43,3 +45,22 @@ data HaddockPrintStyle
...
+
+isDiffFriendly :: ImportExportStyle -> Bool
+isDiffFriendly = (==) ImportExportDiffFriendly
+
+importExportCommaStyle :: ImportExportStyle -> Maybe CommaStyle
+importExportCommaStyle = \case
+  ImportExportLeading -> Just Leading
+  ImportExportTrailing -> Just Trailing
+  ImportExportDiffFriendly -> Nothing
```
Comment (brandonchinn178) [inline src/Ormolu/Config/Types.hs:66]: Shouldnt DiffFriendly just map to Trailing? This also only seems to be used in one location, can we just inline there?

File: src/Ormolu/Printer/Meat/ImportExport.hs [hunk 4]
```diff
@@ -209,8 +210,17 @@ parens' topLevelImport m =
...
     trailingParen = if topLevelImport then txt " )" else txt ")"
 
+getCommaStyle :: R CommaStyle
+getCommaStyle = do
+  style <- getPrinterOpt poImportExportStyle
+  pure $ fromMaybe Trailing $ importExportCommaStyle style
+
+getDiffFriendly :: R Bool
+getDiffFriendly =
+  isDiffFriendly <$> getPrinterOpt poImportExportStyle
```
Comment (brandonchinn178) [inline src/Ormolu/Printer/Meat/ImportExport.hs:220]: this is only used in one location, can we just inline?

**Review by brandonchinn178 — COMMENTED** [review]
Looks great! A few comments; can you also make sure to update the "Specifying config" section in the README too?

File: src/Ormolu/Printer/Meat/Module.hs (hunk 1 above)
Comment (3kyro) [reply src/Ormolu/Printer/Meat/Module.hs:26]: Done, I've also removed an earlier one I've sneaked in `Meat/ImportExport.hs` during the leading commas PR.

File: src/Ormolu/Config/Types.hs (hunk 2 above)
Comment (3kyro) [reply src/Ormolu/Config/Types.hs:57]: Removed

File: src/Ormolu/Config/Types.hs (hunk 3 above)
Comment (3kyro) [reply src/Ormolu/Config/Types.hs]: Yes, it simplifies things a bit. Done.

File: src/Ormolu/Printer/Meat/ImportExport.hs (hunk 4 above)
Comment (3kyro) [reply src/Ormolu/Printer/Meat/ImportExport.hs]: done

File: src/Ormolu/Printer/Meat/ImportExport.hs [hunk 5]
```diff
@@ -192,10 +190,11 @@ parens' topLevelImport m =
       body
       txt ")"
   where
+    getDiffFriendly = isDiffFriendly <$> getPrinterOpt poImportExportStyle
```
Comment (brandonchinn178) [inline src/Ormolu/Printer/Meat/ImportExport.hs:193]: Inline this as well? Should be able to get rid of isDiffFriendly altogether

**Review by brandonchinn178 — COMMENTED** [review]
I think you're still missing the README change I asked about, in the "Specifying configuration" section

File: src/Ormolu/Printer/Meat/ImportExport.hs (hunk 5 above)
Comment (3kyro) [reply src/Ormolu/Printer/Meat/ImportExport.hs]: ok, inlined at both use sites

**Comment by 3kyro** [comment]
> I think you're still missing the README change I asked about, in the "Specifying configuration" section

ah, yes I did :(. I'm not sure why the fourmolu style was mentioning leading commas, but default configuration should be `diff-friendly`.

**Review by brandonchinn178 — APPROVED** [review]
Looks great, thank you!

**Pull request merged**